easybuild/framework/easyconfig/types.py

```diff
@@ -76,6 +78,16 @@ def is_value_of_type(value, typ_spec):
                 # check whether all values have allowed types
                 'value_types': lambda val: all([type(el) in extra_reqs['value_types'] for el in val.values()]),
             }
+        elif parent_type == list:
+            extra_req_checkers = {
+                # check required type
+                'element_types': lambda val: all([type(el) in extra_reqs['element_types'] for el in val]),
```

Comment:
we should flesh out a small function that can check the type of list elements and dict values, and use that for both `value_types` and `element_types` (which should be merged together probably)